base/applications/utilman/process.c

```diff
@@ -44,17 +43,56 @@ BOOL IsProcessRunning(IN LPCWSTR lpProcessName)
         {
             if (_wcsicmp(Process.szExeFile, lpProcessName) == 0)
             {
-                /* The process we are searching for is running */
-                bIsRunning = TRUE;
-                break;
+                /* The names match, return the process ID we're interested */
+                CloseHandle(hSnapshot);
+                return Process.th32ProcessID;
             }
-        }
+        } 
         while (Process32NextW(hSnapshot, &Process));
     }
 
-    /* Free the handle and return */
     CloseHandle(hSnapshot);
-    return bIsRunning;
+    return 0;
+}
+
+/**
+ * @IsProcessRunning
+ *
+ * Checks if a process is running.
+ *
+ * @param[in]   lpProcessName
+ *     The name of the executable process.
+ *
+ * @return
+ *     Returns TRUE if the given process' name is running,
+ *     FALSE otherwise.
+ *
+ */
+BOOL IsProcessRunning(IN LPCWSTR lpProcessName)
+{
+    DWORD dwExitCode, dwProcessID;
+    HANDLE hProcess;
+
+    /* Get the process ID */
+    dwProcessID = GetProcessID(lpProcessName);
+    if (dwProcessID == 0)
+    {
+        return FALSE;
+    }
+
+    /* Query some information from the process in question and check the exit code given by the process */
+    hProcess = OpenProcess(PROCESS_QUERY_INFORMATION, FALSE, dwProcessID);
+    GetExitCodeProcess(hProcess, &dwExitCode);
```

Comment:
`GetExitCodeProcess` can fail (i.e. needs a return value check); but also, it's not the best way to check whether a process is running (what if a process exits with return code `STILL_ACTIVE`? See the documentation).
`WaitForSingleObject` with a zero timeout is a better indication (assuming you in fact get a valid handle, per the comment above).